### DISCUSSION TITLE
Transmission crashes using Curl 8.9.1 or later

### DISCUSSION THREAD

**Discussion by donacarr** [Freetz-NG/freetz-ng#1047] [Allgemein] [answered]
Hi

I just upgraded freetz from tag ng23080 to tag ng24080  and I noted that transmission was updated from 4.0.4 to 4.0.6
Any idea how to troubleshoot this error?
Something like debug log  or whatever?
Thx in advance

**Comment by fda77** [comment]
A common problem is to save files to /tmp or /var which is a part of RAM and when its full the device crashes with OOM
If you forward syslog to another device (or save it to usb storage) you could read last messages after ceash

**Comment by donacarr** [comment]
I started syslogd and now I see one interesting trace:
_**Sep 27 13:39:56 fritz user.err dsld[2427]: internet: Land Attack detected (1)**_

```
Sep 27 15:39:41 fritz kern.info kernel: [202354.987674][1]usb 4-1: usb auto-resume
Sep 27 15:39:41 fritz kern.info kernel: [202355.055457][1]Bus#4 resume: AVM Powermeter changed to 400 mA
Sep 27 15:39:41 fritz kern.info kernel: [202355.055536][1]usb 4-1: finish resume
Sep 27 15:39:42 fritz kern.err kernel: [202355.171088][0]usb-storage 4-1:1.0 - spin up time 208 ms
Sep 27 15:39:44 fritz daemon.debug transmission-daemon[23729]: open-files.cc:47 Fast preallocation failed: Operation not supported (122) (open-files.cc:47)
Sep 27 15:39:47 fritz kern.warn kernel: [202360.376805][2][2]system-load  27% loadavg 2.70 2.39 2.37 - task runtime:19% max:transmission-da 15%  softirqs:2.27% (do_cbm_tasklet 100%) pgstat: sum=109398 free=6545 slab=11555 alloc=285934/s fault=434740/s (sleep 10)
Sep 27 15:39:50 fritz kern.warn kernel: [202363.942465][1][1]system-load  44% loadavg 2.70 2.39 2.37 - task runtime:24% max:transmission-da 21%  softirqs:4.96% (ce_tasklet [qca_ol] 100%) pgstat: sum=106784 free=5178 slab=11610 alloc=285907/s fault=433916/s (sleep 0)
Sep 27 13:39:56 fritz user.err dsld[2427]: internet: Land Attack detected (1)
Sep 27 15:39:57 fritz kern.warn kernel: [202370.378664][2][2]system-load  30% loadavg 2.98 2.46 2.39 - task runtime:3% max:grx-ambient-lig 0%  softirqs:4.36% (do_cbm_tasklet 54%) pgstat: sum=107476 free=6231 slab=11652 alloc=665/s fault=1004/s (sleep 10)
Sep 27 15:40:02 fritz kern.warn kernel: [202375.536687][1][wifi1] FWLOG: [57749940] WHAL_ERROR_RESET_RXFRAME ( 0xbad5, 0x1 )
Sep 27 15:40:21 fritz kern.info kernel: [202394.409126][3]usb 4-1: usb auto-suspend, wakeup 0
Sep 27 15:40:21 fritz kern.info kernel: [202394.409162][3]Bus#4 suspend: AVM Powermeter changed to 50 mA
```


Never saw that in the past!

I am getting the issue with only two recent torrent files...

**Comment by fda77** [comment]
```
 dsld[2427]: internet: Land Attack detected (1)
```

"dsld" is by avm, "land attack" https://www.imperva.com/learn/ddos/land-attacks/

> A LAND Attack is a Layer 4 Denial of Service (DoS) attack in which, the attacker sets the source and destination information of a TCP segment to be the same. A vulnerable machine will crash or freeze due to the packet being repeatedly processed by the TCP stack.

maybe you just misconfigured transmission? (i dont know it)

> I am getting the issue with only two recent torrent files...

or maybe there are bad seeder for this file? 


but this is also interesting:

```
 transmission-daemon[23729]: open-files.cc:47 Fast preallocation failed: Operation not supported (122) (open-files.cc:47)
```

maybe it could be disabled during compilation

have you asked/searched transmission issues/forum/diskussion? https://github.com/transmission/transmission/issues

Also there is a blocklist for some bad ips: https://freetz-ng.github.io/freetz-ng/make/transmission.html

**Comment by donacarr** [comment]
Hi @fda77 

I want to  include transmission 4.0.4 in the tag ng24080 so I basically changed two files:
make/pkgs/transmission/Config.in
make/pkgs/transmission/transmission.mk

to make sure to download 4.0.4 

but when building I got this error:
```

preparing ... mkdir -p source/target-mips_gcc-13.3.0_uClibc-1.0.49-nptl_kernel-4.9/transmission-4.0.4; tools/unxz -c dl/transmission-4.0.4.tar.xz | tools/tar-gnu -x -C source/target-mips_gcc-13.3.0_uClibc-1.0.49-nptl_kernel-4.9/transmission-4.0.4 --transform='s|^./\+||' --strip-components=1
set -e; shopt -s nullglob; for i in make/pkgs/transmission/patches/*.patch* make/pkgs/transmission/patches/current/*.patch*; do case $i in *.patch|*.patch.gz|*.patch.bzip2|*.patch.bz2|*.patch.bz|*.patch.xz|*.patch.lz|*.patch.lzma|*.patch.Z|*.patch.diff) ;; *) continue ;; esac; tools/freetz_patch source/target-mips_gcc-13.3.0_uClibc-1.0.49-nptl_kernel-4.9/transmission-4.0.4 $i ; done;
    applying patch file make/pkgs/transmission/patches/current/020-dont_scrape_paused_torrents_by_default.patch
    patching file libtransmission/session-settings.h
    ----------------------------------------------------------------------
    applying patch file make/pkgs/transmission/patches/current/030-no_in6addr_any_in_libc_without_IPv6_fix.patch
    can't find file to patch at input line 3
    Perhaps you used the wrong -p or --strip option?
    The text leading up to this was:
    --------------------------
    |--- third-party/miniupnp/miniupnpc/src/minissdpc.c
    |+++ third-party/miniupnp/miniupnpc/src/minissdpc.c
    --------------------------
    No file to patch.  Skipping patch.
    1 out of 1 hunk ignored
ERROR: modpatch: Error in patch-file make/pkgs/transmission/patches/current/030-no_in6addr_any_in_libc_without_IPv6_fix.patch
make[1]: *** [make/pkgs/transmission/transmission.mk:141: source/target-mips_gcc-13.3.0_uClibc-1.0.49-nptl_kernel-4.9/transmission-4.0.4/.unpacked] Error 2
make: *** [Makefile:47: envira] Terminated
```

What else I am missing?

**Reply by fda77** [reply]
Have you reported the problem upstream so 4.07 could fix it?
https://github.com/Freetz-NG/freetz-ng/commits/master/make/pkgs/transmission

**Comment by donacarr** [comment]
What do you mean ?
To make sure if 4.0.6 is the issue I want to try 4.0.4  first.
And this is why I want to build a new freetz changing only transmission version.
Do you see my point?

**Comment by fda77** [comment]
why not 4.05?

**Comment by donacarr** [comment]
Since my previous build was based on 7.57 and transmission 4.0.4
I could try 4.0.5 ..but please let me know how to fix the build error above since I got same build error with 4.0.5

**Comment by donacarr** [comment]
@fda77 I did more tests..
I built freetz ng23080 with fw 7.57 and transmission 3.0.0 ---> No issue
Then I just upgraded to freetz ng24080 but using still fw 7.57 and transmission 3.0.0 --> replicated the issue .. Transmission crashed as soon as I add a torrent.

So it is not an issue with transmission but something changed in freetz between ng23080 and ng24080.

![image](https://github.com/user-attachments/assets/81694e0a-3cec-4826-b1b0-8a965bdb0396)

This is my freetz config:

![image](https://github.com/user-attachments/assets/c4630181-26ab-44e2-a9ab-3600cc225414)



Any idea?

**Comment by donacarr** [comment]
@fda77 I did more test and I found the root cause of the issue.
I built freetz using tag ng24080 and **Curl 8.9.0.**
Everything is fine.

But if I use one of the following commit I face again the issue.
![image](https://github.com/user-attachments/assets/5dbb2690-ce71-4f5f-b807-1303391eaf08)

Could you have missed to patch properly latest curl releases?

**Comment by fda77** [comment]
As i wrote above:

> have you asked/searched transmission issues/forum/diskussion? https://github.com/transmission/transmission/issues

if there is some incompatibility they should know

**Comment by donacarr** [comment]
I found the issue mentioned here https://gitlab.alpinelinux.org/alpine/aports/-/issues/16493

curl 8.9.1 needs this patch https://git.alpinelinux.org/aports/commit/main/curl?h=3.20-stable&id=c155d8b610b2944dde558406132df8b3de68aab4

Also discussed here https://github.com/transmission/transmission/issues/7133

**Comment by fda77** [comment]
Thanks, but it is yet changed in freetz!

This is the source of the patch: https://github.com/curl/curl/pull/14390/commits/fe9baa1619d13a1d3ac6bfa530abe229805cddff

Check unpacked source, for my device the directory is:
 `source/target-mips_gcc-13.3.0_uClibc-1.0.50-nptl_kernel-4.9/curl-8.10.1/lib/sigpipe.h`

**Comment by fda77** [comment, marked as answer]
@donacarr maybe new [8.11.0](https://curl.se/ch/8.11.0.html) works?  2d0213f07

**Reply by fda77** [reply]
@donacarr thx for feedback!!1

**Comment by fda77** [comment]
97ad4b7e47fefa2dd9d70ede5023691fa7f8c2ac works?
